src/libraries/Microsoft.Extensions.Diagnostics/tests/ConsoleMetricListenerTests.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.IO;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.Extensions.Diagnostics.Metrics.Tests
+{
+    public class ConsoleMetricListenerTests
+    {
+        [Fact]
+        public void ListenerCanBeRegisteredViaDi()
+        {
+            ServiceCollection services = new ServiceCollection();
+            services.AddMetrics(builder =>
+            {
+                builder.AddDebugConsole();
+                builder.EnableMetrics<ConsoleMetricListener>("TestMeter");
```

Comment:
@noahfalk  `EnableMetrics<T>` is weird, `ConsoleMetricListener` was supposed to be internal. Also there's no overload that takes a listener name.

Reply:
I agree it does feel a little odd. Do you know who it was on the ASP.NET team that designed the original ILogger APIs here that we are copying? It would be nice to get their input if they have any regrets about the design in this area and if they encourage us to replicate it or diverge.

Looking around a little bit in the existing logging design for ILoggingBuilder.AddFilter() I noticed a few things:
- There is a ProviderAliasAttribute on providers that is used to set the name and I missed it. I assume we should add it.
- You can filter by name rather than type if you directly create a Rule object and add it to the Rules list. I still agree its odd that there are no overloads of AddFilter() which take listener name in their parameter list, everything assumes type. 

I'd be fine with modified versions where we used a name instead of the type in overloads, but I am curious to hear from the original authors if possible. @tarekgh - have you heard any user feedback around this aspect of the ILogger design?

Reply:
ProviderAliasAttribute I replaced with IMetricListener.Name. I think they would have done that before if they could have added properties to interfaces in logging.

I don't think the original logging authors are still around.

Reply:
> have you heard any user feedback around this aspect of the ILogger design?

Unfortunately, I don't have any data around that, but I am not a fan of the pattern using the type. So, I hope we can proceed with @Tratcher proposal.